resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -96,12 +96,21 @@ private[spark] class Client(
   } else {
     sparkConf.get(AM_MEMORY).toInt
   }
+
+  private val driverMinimumMemoryOverhead =
+    if (isClusterMode && sparkConf.contains(DRIVER_MIN_MEMORY_OVERHEAD)) {
+      sparkConf.get(DRIVER_MIN_MEMORY_OVERHEAD).get
+    } else {
+      ResourceProfile.MEMORY_OVERHEAD_MIN_MIB
+    }
+
   private val amMemoryOverhead = {
     val amMemoryOverheadEntry = if (isClusterMode) DRIVER_MEMORY_OVERHEAD else AM_MEMORY_OVERHEAD
     sparkConf.get(amMemoryOverheadEntry).getOrElse(
       math.max((amMemoryOverheadFactor * amMemory).toLong,
-        ResourceProfile.MEMORY_OVERHEAD_MIN_MIB)).toInt
+        driverMinimumMemoryOverhead)).toInt
```

Comment:
driver and AM aren't necessarily the same process.  In cluster mode it is but in client mode it isn't (see spark.yarn.am.memoryOverheah) - https://spark.apache.org/docs/latest/running-on-yarn.html.
Unless you have a use case for the app master doing native stuff, which I don't see, I think this should stay the same (384) or if you have a use case for it we should have a separate config for it since its a YARN specific thing.

Reply:
I may be misunderstanding something, but doesn't the fact that above we have 

```
  private val driverMinimumMemoryOverhead =
    if (isClusterMode) {
      sparkConf.get(DRIVER_MIN_MEMORY_OVERHEAD)
    } else {
      ResourceProfile.MEMORY_OVERHEAD_MIN_MIB
    }
```
    
 guarantee that my changes are only applied in cluster mode, which, form that you said, seems to be the case where it actually corresponds to the driver?

Reply:
ah yes that works